### PR TITLE
[tanks] Mark non-validated fight styles.

### PR DESCRIPTION
**Pull request by renanthera**
Marks Dungeon Slice and Dungeon route as non-validated fight styles for all tanks, until at minimum tank damage intake defaults are fixed.

### CODE REVIEW

File: engine/class_modules/monk/sc_monk.cpp [hunk 1]
```diff
@@ -7750,6 +7750,22 @@ namespace monk
...
+      switch ( style )
+      {
+      case FIGHT_STYLE_DUNGEON_ROUTE:
+      case FIGHT_STYLE_DUNGEON_SLICE:
+        return false;
+      default:
+        return true;
+      }
+    }
+    return true;
```
Comment (Hinalover) [inline engine/class_modules/monk/sc_monk.cpp:7779]: I know this is for Brewmaster, but for Windwalker, Patchwork, Casting Patchwork, Dungeon Route and Dungeon Slice are the only ones that are valid for us. Add Cleave and Hectic Add Cleave are not supported due to the APL not holding CDs for add spawning. Also any movement is not supported as well.

File: engine/class_modules/paladin/sc_paladin.cpp [hunk 2]
```diff
@@ -2524,6 +2524,23 @@ void paladin_t::init_action_list()
...
+  if ( specialization() == PALADIN_PROTECTION )
+  {
+    switch ( style )
+    {
+      case FIGHT_STYLE_DUNGEON_ROUTE:
+      case FIGHT_STYLE_DUNGEON_SLICE:
+        return true;
+      default:
+        return false;
+    }
```
Comment (msserrano) [inline engine/class_modules/paladin/sc_paladin.cpp:2539]: This seems backwards - I think you have true / false flipped here

File: engine/class_modules/paladin/sc_paladin.cpp [hunk 3]
```diff
@@ -2524,6 +2524,23 @@ void paladin_t::init_action_list()
...
+  }
+  return true;
+}
+
```
Comment (mserrano) [inline engine/class_modules/paladin/sc_paladin.cpp]:
```suggestion
bool paladin_t::validate_fight_style( fight_style_e style ) const
{
  if ( specialization() == PALADIN_PROTECTION )
  {
    switch ( style )
    {
      case FIGHT_STYLE_DUNGEON_ROUTE:
      case FIGHT_STYLE_DUNGEON_SLICE:
        return false;
      default:
        return true;
    }
  }
  return true;
}

```

**Review by mserrano — CHANGES_REQUESTED** [review]

File: engine/class_modules/sc_death_knight.cpp [hunk 4]
```diff
@@ -10405,6 +10406,24 @@ void death_knight_t::init_finished()
...
+    {
+    case FIGHT_STYLE_DUNGEON_ROUTE:
+    case FIGHT_STYLE_DUNGEON_SLICE:
+      return false;
+    default:
+      return true;
+    }
+  }
+  return true;
+}
```
Comment (Khazakdk) [inline engine/class_modules/sc_death_knight.cpp]:
```suggestion
bool death_knight_t::validate_fight_style( fight_style_e fight ) const
{
  if ( specialization() == DEATH_KNIGHT_BLOOD )
  {
    switch ( fight )
    {
    case FIGHT_STYLE_PATCHWERK:
    case FIGHT_STYLE_CASTING_PATCHWERK:
      return true;
    default:
      return false;
    }    
  }
  if ( specialization() == DEATH_KNIGHT_FROST )
  {
    switch ( fight )
    {
    case FIGHT_STYLE_PATCHWERK:
    case FIGHT_STYLE_CASTING_PATCHWERK:
    case FIGHT_STYLE_DUNGEON_SLICE:
    case FIGHT_STYLE_DUNGEON_ROUTE:
      return true;
    default:
      return false;
    }
  }

  if ( specialization() == DEATH_KNIGHT_UNHOLY )
  {
    switch ( fight )
    {
    case FIGHT_STYLE_PATCHWERK:
    case FIGHT_STYLE_CASTING_PATCHWERK:
    case FIGHT_STYLE_DUNGEON_SLICE:
    case FIGHT_STYLE_DUNGEON_ROUTE:
      return true;
    default:
      return false;
    }
  }
  return false;
}
```

I'd like to leverage `validate_fight_style` for DPS specs as well and I think the easiest way would be to drop the new function into your PR.

File: engine/class_modules/paladin/sc_paladin.cpp (hunk 3 above)
Comment (renanthera) [reply engine/class_modules/paladin/sc_paladin.cpp]: Manually corrected for in conjunction with combining changes from #8665.

File: engine/class_modules/sc_death_knight.cpp (hunk 4 above)
Comment (renanthera) [reply engine/class_modules/sc_death_knight.cpp]: Manually corrected for paladin in conjunction with combining changes from https://github.com/simulationcraft/simc/pull/8665.

**Pull request merged**